dotcom-rendering/src/footballTables.ts

```diff
@@ -1,36 +1,150 @@
+	points: number;
+	results: TeamResult[];
 };
 
 export type FootballTableCompetition = {
 	url: string;
 	name: string;
 	hasGroups: boolean;
+	dividers: number[];
 	tables: FootballTableData[];
 };
 
 export type FootballTableData = {
-	groupName: string;
-	dividers: number[];
-	entries: {
-		position: number;
-		team: { name: string; id: string; url: string };
-		gamesPlayed: number;
-		won: number;
-		drawn: number;
-		lost: number;
-		goalsFor: number;
-		goalsAgainst: number;
-		goalDifference: number;
-		points: number;
-		results: TeamResult[];
-	}[];
-	linkToFullTable: boolean;
+	groupName?: string;
+	entries: Entry[];
+};
+
+export type FootballTableCompetitions = FootballTableCompetition[];
+
+const parseTables = (
+	feGroups: FEFootballTable['groups'],
+): Result<ParserError, FootballTableData[]> => {
+	const groupsParser = listParse(parseTable);
+	return groupsParser(feGroups);
+};
+
+const parseTable = (
+	feGroup: FEGroup,
+): Result<ParserError, FootballTableData> => {
+	const entriesParser = listParse(parseEntry);
+	const parsedEntries = entriesParser(feGroup.entries);
+
+	if (parsedEntries.kind === 'error') {
+		return parsedEntries;
+	}
+
+	return ok({
+		groupName: feGroup.round.name,
+		entries: parsedEntries.value.sort((a, b) => a.position - b.position),
+	});
+};
+
+const parseResult = (result: FETeamResult): Result<ParserError, TeamResult> => {
+	return ok({
+		matchId: result.matchId,
+		self: {
+			id: result.self.id,
+			name: result.self.name,
+			score: result.self.score,
+		},
+		foe: {
+			id: result.foe.id,
+			name: result.foe.name,
+			score: result.foe.score,
+		},
+	});
+};
+
+const parseResults = (
+	teamResults: FETeamResult[],
+): Result<ParserError, TeamResult[]> => {
+	const resultsParser = listParse(parseResult);
+	return resultsParser(teamResults);
 };
+
+const parseEntry = (
+	feEntry: FELeagueTableEntry,
+): Result<ParserError, Entry> => {
+	const { team, teamUrl } = feEntry;
+
+	const parsedResults = parseResults(feEntry.results);
+
+	if (parsedResults.kind === 'error') {
+		return parsedResults;
+	}
+
+	return ok({
+		position: team.rank,
+		team: {
+			name: team.name,
+			id: team.id,
+			url: teamUrl,
+		},
+		gamesPlayed: team.total.played,
+		won: team.total.won,
+		drawn: team.total.drawn,
+		lost: team.total.lost,
+		goalsFor: team.total.goalsFor,
+		goalsAgainst: team.total.goalsAgainst,
+		goalDifference: team.goalDifference,
+		points: team.points,
+		results: parsedResults.value,
+	});
+};
+
+const parseFootballTables = (
+	table: FEFootballTable,
+): Result<ParserError, FootballTableCompetition> => {
+	const parsedTables = parseTables(table.groups);
+
+	if (parsedTables.kind === 'error') {
+		return parsedTables;
+	}
+
+	return ok({
+		url: table.competition.url,
+		name: table.competition.fullName,
+		hasGroups: table.hasGroups,
+		dividers: table.competition.dividers ?? [],
+		tables: parsedTables.value,
+	});
+};
+
+// ToDo: if we don't return any errors can we remove the Result type?
```

Comment:
I left this comment because it seems odd to be returning a `Result` type when we are never returning errors in our parsing. Initially we wrote it as a parser because we assumed there would be some parsing errors to return. What do you think @JamieB-gu @marjisound ?

Reply:
Good question. I think to be consistent with other parser usages, it might still be good to return Result even if in practice the parser won't error. But I'm interested to see what @JamieB-gu  thinks too

Reply:
This sounds reasonable. If you're just transforming the data rather than parsing it then you don't need `Result`.

Reply:
I think it might be better to do this in a separate PR as this PR has gone huge and it's been fully tested. I created this [ticket ](https://github.com/orgs/guardian/projects/88/views/15?sliceBy%5Bvalue%5D=Decommission+Frontend+Rendering&pane=issue&itemId=105511587&issue=guardian%7Cdotcom-rendering%7C13752). And I'm happy to do it after this PR or maybe leave it to after the health sprint. 

Let me know if any of you object!